### DISCUSSION TITLE
Not working Create 3 lightring by ros2 topic

### DISCUSSION THREAD

**Discussion by seungin-cha** [iRobotEducation/create3_docs#587] [Questions]
Hello, I'm a turtlebot4 user.

I'm trying to do turtlebot4 tutorials, and I found one problem not working.
Create 3 does not respond to ros2 topic `/cmd_lightring` publishing.
I'm using this same code, https://turtlebot.github.io/turtlebot4-user-manual/tutorials/first_node_cpp.html.
So I cannot see light color changing even if the publishing data has `override_system=true`.

This is my try:

```shell
$ ros2 run turtlebot4_cpp_tutorials turtlebot4_cpp_first_node
[...] [...]: Button 1 Pressed!
[...] [...]: Button 1 Pressed!
[...] [...]: Button 1 Pressed!
...
```

Interface butons respond to clicks normally.
And ROS2 communication is good among User PC, RPi4, and Create 3.

This is my diagnositics result:
It seems no problem about lightring control.

![image](https://github.com/iRobotEducation/create3_docs/assets/147928987/a6961591-7a7e-478c-a67c-ce0e116bd954)

Thank you!

**Comment by alsora** [comment]
Hi @seungin-cha,

> And ROS2 communication is good among User PC, RPi4, and Create 3.

I assume that this means that you are able to drive the robot around and do all other operations, with only the lightring failing to work.

I suspect that this could be caused by lack of reliability in your communication.
i.e. you are sending a single message, but this gets dropped by the network interface.

Could you try the following?
Send repeated lightring messages 
```
ros2 topic pub /cmd_lightring irobot_create_msgs/msg/LightringLeds "{override_system: true, leds: [{red: 255, green: 0, blue: 0}, {red: 0, green: 255, blue: 0}, {red: 0, green: 0, blue: 255}, {red: 255, green: 255, blue: 0}, {red: 255, green: 0, blue: 255}, {red: 0, green: 255, blue: 255}]}" -r 10
```

Send lightring messages with reliable QoS
```
ros2 topic pub /cmd_lightring irobot_create_msgs/msg/LightringLeds "{override_system: true, leds: [{red: 255, green: 0, blue: 0}, {red: 0, green: 255, blue: 0}, {red: 0, green: 0, blue: 255}, {red: 255, green: 255, blue: 0}, {red: 255, green: 0, blue: 255}, {red: 0, green: 255, blue: 255}]}" --qos-reliability reliable
```

If these don't work, then my assumption is likely wrong and the problem could be elsewhere.
It would be great to get a log from the create 3 while you are sending the commands.
You can find the log in the webserver: https://iroboteducation.github.io/create3_docs/webserver/logs/

Try also to use the ROS 2 action to control lightrings
```
ros2 action send_goal /led_animation irobot_create_msgs/action/LedAnimation "{animation_type: 2,max_runtime:{sec: 10,nanosec: 0},lightring:{override_system: true, leds: [{red: 255, green: 0, blue: 0}, {red: 0, green: 255, blue: 0}, {red: 0, green: 0, blue: 255}, {red: 255, green: 255, blue: 0}, {red: 255, green: 0, blue: 255}, {red: 0, green: 255, blue: 255}]}}"
```
